lib/bumblebee/layers/transformer.ex

```diff
@@ -187,6 +214,12 @@ defmodule Bumblebee.Layers.Transformer do
     * `:kernel_initializer` - initializer for kernel weights. Defaults
       to `:glorot_uniform`
 
+    * `:projection_size` - the projection size for key, value, and query
+      states per-head. In most cases, the projection size is equal to
+      `hidden_size // num_heads`. However, if set, this will project key,
+      value, and query by a factor of `projection_size * num_heads`.
+      Defaults to `nil`
```

Comment:
?

```suggestion
    * `:attention_projection_size` - the projection size for key, value,
      and query states per-head. Defaults to `div(hidden_size, num_attention_heads)`
```

I would also place it below `:hidde_size` as other sizes, but just cosmetics :)